src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp

```diff
@@ -212,132 +213,39 @@ CorporealityEntry const _corporealityReference[MAX_CORPOREALITY_STATE] = {
     {74831, 74836}
 };
 
-struct generic_halionAI : public BossAI
```

Comment:
then you just override EnterCombat in children AIs and set the timers there (after calling parent's)